### DISCUSSION TITLE
History of test per tentant (country) (or any other parameter)

### DISCUSSION THREAD

**Discussion by dnkcz** [allure-framework/allure2#2433] [General Discussion]
Hi, we have same test for each country. In allure report in history we need to distinguish between results of each country e.g. to se only results for that country. I tryied to add Allure.parameter(country, country, false, Parameter.Mode.DEFAULT); but this does not help because it does not separate history for each country for same test. Any ideas appreciated. Thanks.

**Comment by dnkcz** [comment]
basically i need to modify historyId so that it takes into account some parameters. We are using JUNIT5.

**Reply by baev** [reply]
The issue https://github.com/allure-framework/allure-java/issues/349

**Reply by baev** [reply]
As a workaround, you may use `TestLifecycleListener` and recalculate `historyId` at the end of the test.

**Reply by dnkcz** [reply]
thanks a lot for quick answer. I already figured out the same things, so good to have solution confirmed. TY :)
